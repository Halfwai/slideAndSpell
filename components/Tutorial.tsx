import { View, Text, StyleSheet, Animated, Dimensions } from 'react-native';

import GameBoard from "@/components/Gameboard";
import {COLOURS} from "@/constants/colours";

import { MaterialCommunityIcons } from '@expo/vector-icons';

import { useRef, useEffect, useState } from 'react';
import MyAppText from '@/components/MyAppText';
import AuthButton from '@/components/AuthButton';



interface TutorialProps {
    endTutorial: Function
}

export default function Tutorial(props: TutorialProps) {
    const width = Dimensions.get("window").width;
    const [pointerStartPos, setPointerStartPos] = useState({x: (width * 0.85) / 3, y: 0});
    const [pointerEndPos, setPointerEndPos] = useState({x: (width * 0.85) / 3, y: -((width * 0.85) / 3)});
    const [instructions, setInstructions] = useState<string>("Slide the tiles to form words");


    const [showPointer, setShowPointer] = useState(true);


    const pointerPosition = useRef(new Animated.ValueXY({ x: pointerStartPos.x, y: pointerStartPos.y })).current;
    const pointerOpacity = useRef(new Animated.Value(0)).current;


    const [completeWords, setCompleteWords] = useState<number>(0);
    const [restartAnimation, setRestartAnimation] = useState<boolean>(false);


    useEffect(() => {
        setRestartAnimation(!restartAnimation); 
        if(completeWords === 0){
            setPointerStartPos({
                x: (width * 0.85) / 3,
                y: 0
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: -((width * 0.85) / 3)
            })
            setInstructions("Slide the tiles to form words");
        };
        if (completeWords === 1){
            setPointerStartPos({
                x: (width * 0.85) / 3,
                y: (width * 0.85) / 3
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: 0
            })
            setInstructions("Slide the tiles to form words");
        };
        if (completeWords === 2){
            setPointerStartPos({
                x: 0,
                y: -((width * 0.85) / 3) * 2
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: (width * 0.85) / 3
            })
            setInstructions("When one word remains, slide the final tile to complete the puzzle");
        };
            
    }, [completeWords]);  


    useEffect(() => {
        animatePointer();
    }, []);

    useEffect(() => {   
        resetPointer();
    }, [restartAnimation]);

    function resetPointer() {
        Animated.sequence([
            Animated.timing(pointerOpacity, {
                toValue: 0,
                duration: 0,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerStartPos.x, y: pointerStartPos.y },
                duration: 0,
                useNativeDriver: true
            }),
        ]).start((finished) => {
            if(finished){
                animatePointer();
            }
        });
    }

    function animatePointer() {
        Animated.sequence([
            Animated.timing(pointerOpacity, {
                toValue: 1,
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerEndPos.x, y: pointerEndPos.y },
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerOpacity, {
                toValue: 0,
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerStartPos.x, y: pointerStartPos.y },
                duration: 0,
                useNativeDriver: true
            }),
        ]).start(({finished}) => {
            if(finished){
                animatePointer();
            }
        });
    }


    return (
        <View
            style={{
                flex: 1,
                justifyContent: "center",
                alignItems: "center",
                backgroundColor: "#CBE6F7",
                width: "100%",
            }}
        >
            {showPointer &&
                <View
                    style={{
                        position: "absolute",
                        top: 20,
                        justifyContent: "center",
                        alignItems: "center",
                        height: 100
                    }}
                >
                    <MyAppText
                        style={{
                            fontSize: 25,
                            textAlign: "center",

                        }}
                    >
                        {instructions}
                    </MyAppText> 

                </View>

            }
           
            <View
                style={{
                    justifyContent: "center",
                    alignItems: "center",
                    backgroundColor: COLOURS.blue
            }}>
                {showPointer && 
                    <Animated.View style={{position: "absolute",zIndex: 2, justifyContent: "center", alignItems: "center", transform: [{translateX: pointerPosition.x}, {translateY: pointerPosition.y}], opacity: pointerOpacity}}>
                        <MaterialCommunityIcons name="cursor-pointer" size={40} color="#555"/> 
                    </Animated.View>
                }

                <GameBoard 
                    gameBoard={[["t", "w","0"], ["t", "i", "o"],["l", "o", "e"]]}
                    extraLetter={"t"}
                    onGameEnd={() => {
                        setShowPointer(false);
                    }}
                    returnCompleteWords={(count : number) => {
                        setCompleteWords(count);
                    }}
                />
            </View>
            <AuthButton 
                text='End Tutorial'
                onPress={() => {
                    setShowPointer(false);
                    props.endTutorial();
                }}
                style={{
                    backgroundColor: COLOURS.green,
                    position: "absolute",
                    bottom: 20
                }}

            />
        </View>
    )
}   

const styles = StyleSheet.create({

});