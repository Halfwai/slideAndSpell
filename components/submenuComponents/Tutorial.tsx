import React, { useRef, useEffect, useState } from 'react';
import { View, StyleSheet, Animated, Dimensions } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

// Import COLOURS
import { COLOURS } from "@/constants/colours";
import { SIZES } from '@/constants/sizes';

// Import Components
import GameBoard from "@/components/gameComponents/GameBoard";
import MyAppText from '@/components/common/MyAppText';
import AuthButton from '@/components/buttons/AuthButton';

// Setup props
interface TutorialProps {
    endTutorial: Function
}

export default function Tutorial(props: TutorialProps) {
    // Calculate the width of the screen
    const width = Dimensions.get("window").width;
    // Setup state for the pointer
    const [pointerStartPos, setPointerStartPos] = useState({ x: SIZES.GAMEBOARD / 3, y: 0 });
    const [pointerEndPos, setPointerEndPos] = useState({ x: SIZES.GAMEBOARD / 3, y: -(SIZES.GAMEBOARD / 3) });
    const [instructions, setInstructions] = useState<string>("Slide the tiles to form words");
    const [showPointer, setShowPointer] = useState(true);
    const pointerPosition = useRef(new Animated.ValueXY({ x: pointerStartPos.x, y: pointerStartPos.y })).current;
    const pointerOpacity = useRef(new Animated.Value(0)).current;

    // Setup state for the tutorial
    const [completeWords, setCompleteWords] = useState<number>(0);
    const [restartAnimation, setRestartAnimation] = useState<boolean>(false);

    // Update the pointer position based on the number of complete words. This shows the user what to do next
    useEffect(() => {
        setRestartAnimation(!restartAnimation);
        if (completeWords === 0) {
            setPointerStartPos({
                x: (width * 0.85) / 3,
                y: 0
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: -((width * 0.85) / 3)
            })
            setInstructions("Slide the tiles to form words");
        };
        if (completeWords === 1) {
            setPointerStartPos({
                x: (width * 0.85) / 3,
                y: (width * 0.85) / 3
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: 0
            })
            setInstructions("Slide the tiles to form words");
        };
        if (completeWords === 2) {
            setPointerStartPos({
                x: 0,
                y: -((width * 0.85) / 3) * 2
            })
            setPointerEndPos({
                x: (width * 0.85) / 3,
                y: (width * 0.85) / 3
            })
            setInstructions("When one word remains, slide the final tile to complete the puzzle");
        };
    }, [completeWords]);

    // Start the pointer animation
    useEffect(() => {
        animatePointer();
    }, []);

    // Reset the pointer when the animation is restarted
    useEffect(() => {
        resetPointer();
    }, [restartAnimation]);

    // Function to return the pointer to the start position
    function resetPointer() {
        Animated.sequence([
            Animated.timing(pointerOpacity, {
                toValue: 0,
                duration: 0,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerStartPos.x, y: pointerStartPos.y },
                duration: 0,
                useNativeDriver: true
            }),
        ]).start((finished) => {
            if (finished) {
                animatePointer();
            }
        });
    }

    // Function to animate the pointer
    function animatePointer() {
        Animated.sequence([
            Animated.timing(pointerOpacity, {
                toValue: 1,
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerEndPos.x, y: pointerEndPos.y },
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerOpacity, {
                toValue: 0,
                duration: 500,
                useNativeDriver: true
            }),
            Animated.timing(pointerPosition, {
                toValue: { x: pointerStartPos.x, y: pointerStartPos.y },
                duration: 0,
                useNativeDriver: true
            }),
        ]).start(({ finished }) => {
            if (finished) {
                animatePointer();
            }
        });
    }

    return (
        <View style={styles.container}>
            {showPointer &&
                <View style={styles.instructionsContainer}>
                    <MyAppText style={styles.instructionsText}>
                        {instructions}
                    </MyAppText>
                </View>
            }
            <View style={styles.pointerContainer}>
                {showPointer &&
                    <Animated.View style={[styles.pointer, { transform: [{ translateX: pointerPosition.x }, { translateY: pointerPosition.y }], opacity: pointerOpacity }]} pointerEvents="none">
                        <MaterialCommunityIcons name="cursor-pointer" size={40} color="#555" />
                    </Animated.View>
                }
                <GameBoard
                    gameBoard={[["t", "w", "0"], ["t", "i", "o"], ["l", "o", "e"]]}
                    extraLetter={"t"}
                    onGameEnd={() => {
                        setShowPointer(false);
                    }}
                    returnCompleteWords={(count: number) => {
                        setCompleteWords(count);
                    }}
                />
            </View>
            <AuthButton
                text='End Tutorial'
                onPress={() => {
                    setShowPointer(false);
                    props.endTutorial();
                }}
                style={styles.exitButton}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "#CBE6F7",
        width: "100%",
    },
    instructionsContainer: {
        position: "absolute",
        top: 20,
        justifyContent: "center",
        alignItems: "center",
        height: 100
    },
    instructionsText: {
        fontSize: 25,
        textAlign: "center",
    },
    pointerContainer: {
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: COLOURS.blue
    },
    pointer: {
        position: "absolute",
        zIndex: 2,
        justifyContent: "center",
        alignItems: "center",
    },
    exitButton: {
        backgroundColor: COLOURS.green,
        position: "absolute",
        bottom: 20
    }
});