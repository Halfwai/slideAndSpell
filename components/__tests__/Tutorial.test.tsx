import React from 'react';
import { render } from '@testing-library/react-native';
import Tutorial from '@/components/submenuComponents/Tutorial';

it(`Tutorial Component renders correctly`, () => {
    const { toJSON } = render(
        <Tutorial
            endTutorial={() => {}}
        />
    );    
    expect(toJSON()).toBeTruthy();
});