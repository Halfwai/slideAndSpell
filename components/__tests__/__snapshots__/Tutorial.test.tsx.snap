// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#CBE6F7",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
        "position": "absolute",
        "top": 20,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "fontSize": 25,
            "textAlign": "center",
          },
        ]
      }
    >
      Slide the tiles to form words
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CBE6F7",
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 0,
          "position": "absolute",
          "transform": [
            {
              "translateX": 212.5,
            },
            {
              "translateY": 0,
            },
          ],
          "zIndex": 2,
        }
      }
    >
      MaterialCommunityIconsMock
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 637.5,
          "position": "absolute",
          "transform": [
            {
              "translateY": 1334,
            },
          ],
          "width": 637.5,
          "zIndex": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "position": "absolute",
            "transform": [
              {
                "translateX": -750,
              },
              {
                "translateY": -150,
              },
            ],
            "zIndex": 1,
          }
        }
      />
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            t
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 211.83333333333334,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            w
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "transform": [
              {
                "translateX": 423.6666666666667,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 207.83333333333334,
            "zIndex": -1,
          }
        }
      />
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 211.83333333333334,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            t
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 211.83333333333334,
              },
              {
                "translateY": 211.83333333333334,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            i
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 423.6666666666667,
              },
              {
                "translateY": 211.83333333333334,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            o
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 423.6666666666667,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            l
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 211.83333333333334,
              },
              {
                "translateY": 423.6666666666667,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            o
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "backgroundColor": "rgba(170, 170, 170, 1)",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "margin": 1,
            "position": "absolute",
            "transform": [
              {
                "translateX": 423.6666666666667,
              },
              {
                "translateY": 423.6666666666667,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            e
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "position": "absolute",
          "top": -529.5833333333334,
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <View
        collapsable={false}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#AAA",
            "borderRadius": 10,
            "height": 207.83333333333334,
            "justifyContent": "center",
            "margin": 1,
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 207.83333333333334,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Postino_std",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          >
            t
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginTop": 20,
          "position": "absolute",
          "zIndex": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#8BF0BC",
            "borderBottomWidth": 5,
            "borderColor": "white",
            "borderTopWidth": 5,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingVertical": 10,
            "transform": [
              {
                "translateX": -750,
              },
              {
                "translateY": 200,
              },
            ],
            "width": 375,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Time
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          00:00:00
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#8BF0BC",
            "borderBottomWidth": 5,
            "borderColor": "white",
            "borderTopWidth": 5,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingVertical": 10,
            "transform": [
              {
                "translateX": 750,
              },
              {
                "translateY": 200,
              },
            ],
            "width": 375,
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Slides
        </Text>
        <Text
          style={
            {
              "color": "black",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#8BF0BC",
        "borderRadius": 10,
        "borderWidth": 1,
        "bottom": 20,
        "marginBottom": 20,
        "opacity": 1,
        "padding": 12,
        "position": "absolute",
        "width": "90%",
      }
    }
    testID="authButton"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "color": "black",
            "fontSize": 20,
            "fontWeight": "bold",
          },
        ]
      }
    >
      End Tutorial
    </Text>
  </View>
</View>
`;
